test_crates/exhaustive_struct_added/new/src/lib.rs

```diff
@@ -0,0 +1,27 @@
+#![no_std]
+
+// Existing struct should not trigger the lint
+pub struct ExistingStruct {
+    pub bar: u64,
+}
+
+// New non-exhaustive struct with public fields should trigger the lint
+pub struct NewStruct {
+    pub bar : u64,
```

Comment:
nit: please try to run `cargo fmt` on the test crates
```suggestion
    pub bar: u64,
```